app/components/task/details/task-dependencies.component.ts

```diff
@@ -0,0 +1,145 @@
+import { Component, Input, OnDestroy } from "@angular/core";
+
+import { Task, TaskDependency } from "app/models";
+import { TaskService } from "app/services";
+import { BehaviorSubject } from "rxjs";
+
+@Component({
+    selector: "bex-task-dependencies",
+    templateUrl: "task-dependencies.html",
+})
+export class TaskDependenciesComponent implements OnDestroy {
+    @Input()
+    public set jobId(value: string) {
+        this._jobId = (value && value.trim());
+    }
+    public get jobId() { return this._jobId; }
+
+    @Input()
+    public set task(task: Task) {
+        this._task = task;
+        this._refresh(task);
+    }
+    public get task() { return this._task; }
+
+    public dependentIds: string[] = [];
+    public dependencies: BehaviorSubject<TaskDependency[]>;
+    public loadingMore: boolean = false;
+    public hasMore: boolean = false;
+
+    private _jobId: string;
+    private _task: Task;
+    private _skip: number;
+    private _take: number;
+
+    constructor(
+        private taskService: TaskService) {
+    }
+
+    public loadMore() {
+        if (this.dependentIds.length > 0) {
+            const endIndex = this._skip + Math.min(this._take, this.dependentIds.length - this._skip);
+            const taskIdSet = this.dependentIds.slice(this._skip, endIndex);
+            const currentPage = taskIdSet.map(id => {
+                return new TaskDependency(id);
+            });
+
+            this.dependencies.next(this.dependencies.value.concat(currentPage));
+            this.taskService.getMultiple(this.jobId, taskIdSet).subscribe({
+                next: (response: any) => {
+                    if (response.data) {
+                        this._processGetMultipleResponse(response, currentPage);
+                    }
+                },
```

Comment:
I would let the task service handle that. The getMultiple should return the TaskObjects not the raw data